src/PolicyTool/Program.cs

```diff
@@ -0,0 +1,307 @@
+                }, appName, clientSecret);
+
+            addRoles.SetHandler(
+                (string appName, string[] roles) =>
+                {
+                    var policy = appRm.GetPolicies(appName).First(); //There should only be one as multiple policies is not currently supported
+                    foreach (var role in roles)
+                    {
+                        mainBus.TrySend(new ApplicationMsgs.CreateRole(Guid.NewGuid(), role, policy.PolicyId, policy.ApplicationId), out var _);
+                    }
+
+                }
+                , appName, roles);
+            addUser.SetHandler(
+                (string appName, string userName, string domain, string[] roles) =>
+                {
+                    Guid userId;
+                    Guid subjectId;
+
+                    if (validation.TryFindUserPriciple(domain, userName, out UserPrincipal user))
+                    {
+                        if (!userRm.HasUser(user.Sid.Value, user.Context.Name, out userId))
+                        {
+                            userId = Guid.NewGuid();
+                            userStore.AddUser(user, user.Context.Name, user.ContextType.ToString(), userId, out subjectId);
+                            Console.WriteLine($"User added {user.Context.Name}/{userName}");
+                        }
+                        else
+                        {
+                            subjectsRm.TryGetSubjectIdForPrinciple(new PrincipleWrapper(user), out subjectId);
+                            Console.WriteLine($"User found {user.Context.Name}/{userName}");
+                        }
+
+                        Console.WriteLine($"\t UserId    { userId}");
+                        Console.WriteLine($"\t SubjectId { subjectId}");
+                    }
+                    else
+                    {
+                        Console.WriteLine($"user {domain}/{userName} not found.");
+
+                        return;
+                    }
+                    var n = 0;
+                    //Add client scope to user for access
+                    while (!userRm.UsersById.ContainsKey(userId))
+                    {
+                        Thread.Sleep(30);
+                        n++;
+                        //todo: fix the readmodel race condition
+                        if (n % 10 == 0)
+                        {
+                            userRm.Dispose();
+                            userRm = new UsersRm(EsConnection);
+                        }
+
+                    }
+                    var userDto = userRm.UsersById[userId];
+
+                    if (userDto.Scopes.Contains(appName.ToUpper()))
+                    {
+                        Console.WriteLine($"User {user.Context.Name}/{userName} has access to App {appName} , exiting.");
+                        return;
+                    }
+                    if (mainBus.TrySend(new UserMsgs.AddClientScope(userId, appName), out var _))
```

Comment:
Let's make this a correlated message
```suggestion
                    if (mainBus.TrySend(MessageBuilder.New(() => new UserMsgs.AddClientScope(userId, appName), out var _)))
```

Reply:
that means updating the whole chain down, I'll add a todo